wf-galleon/WFLY-19021-Stability_In_Provisioning.adoc

```diff
@@ -0,0 +1,252 @@
+If no stability level is specified, the `default` level is used.
+Stability level is set thanks to a new option of the WildFly Galleon Plugins Maven plugin: 
+`<package-stability>default|community|preview|experimental</package-stability>`
+
+This level can be overridden at provisioning time with the Galleon option `<package-stability>`
+
+
+==== JBoss Modules modules stability level
+
+A stability level can be set in `module.xml` thanks to the property `jboss.stability`. 
+Possible value being `default|community|preview|experimental`. If no stability is specified, `default` is assumed.
+
+==== Implementation notes
+
+At build time, the minimum stability level is used in 2 cases:
+
+* To start the embedded server that generates the Galleon features at the correct stability level (allowing the features to be discovered in the server).
+* To filter-out packages at a stability level lower than the minimum-stability level. Such packages are not packaged inside the feature-pack.
+
+==== Stability levels for WildFly feature-packs
+
+Expected minimum stability levels for WildFly feature-packs:
+
+* wildfly-ee-galleon-pack: `experimental`
+* wildfly-galleon-pack: `experimental`
+* wildfly-preview-feature-pack: `experimental`
+* wildfly-core: `experimental`
+
+Expected config stability levels for WildFly feature-packs:
+
+* wildfly-ee-galleon-pack: `community`
+* wildfly-galleon-pack: `community`
+* wildfly-preview-feature-pack: `preview`
+* wildfly-core: `community`
+
+Expected package stability levels for WildFly feature-packs:
+
+* wildfly-ee-galleon-pack: `community`
+* wildfly-galleon-pack: `community`
+* wildfly-preview-feature-pack: `preview`
+* wildfly-core: `community`
+
+=== Requirements at provisioning time
+
+Provisioning time follows some rules to properly handle stability and avoid features/packages of wrong stability being provisioned.
+
+* The feature-pack contained config stability level is used to constrain the set of provisioned features. Any features at a lower stability level contained in the 
+feature-pack (if any), are not provisioned.
+* The feature-pack contained package stability level is used to constrain the set of provisioned packages. Any packages at a lower stability level contained in the 
+feature-pack (if any), are not provisioned.
+* By default, the config and package stability levels of each feature-pack are used when provisioning its own content. 
+* Each feature-pack present in the provisioning configuration can have a different config stability level. 
+The config stability of each feature-pack applies to its own content and doesn't leak into other feature-packs.
+* Each feature-pack present in the provisioning configuration can have a different package stability level. 
+The package stability of each feature-pack applies to its own content and doesn't leak into other feature-packs.
+* A user can specify a config stability-level, by means of the `config-stability-level` Galleon option. 
```

Comment:
@spyrkob , FYI, we are changing stability-level by 2 options: config-stability-level and package-stability-level.

Reply:
Thanks, created https://github.com/wildfly-extras/prospero/issues/600 for this